### DISCUSSION TITLE
Event Conditions: Player/NPC has Status Effect...

### DISCUSSION THREAD

**Discussion by LunamOdium** [AscensionGameDev/Intersect-Engine#1731] [Ideas]
Would be nice to be able to see if an NPC or a player has a specific status effect(blind, sleep, snare, etc), this could pave way for more event-based features like:

Diseases(vampires/werewolves anyone?)
more intricate spells(soul trap? hehe :D)
unique quests(kinda like Pyrite's questline in Skyrim)
hospitals where they only heal you IF your afflicted(cant waste those precious tax coins lol) 

and probably more I'm not thinking of.

**Comment by Arufonsu** [comment]
> Would be nice to be able to see if an NPC or a player has a specific status effect

? ...
you actually can, just select the NPC/Player you wish to check and you will see their status effect in the entity target box.

Also, status induced events... that's a really bad a idea unless you want to fry your server up in fire.


> Diseases(vampires/werewolves anyone?) more intricate spells(soul trap? hehe :D) unique quests(kinda like Pyrite's questline in Skyrim) hospitals where they only heal you IF your afflicted(cant waste those precious tax coins lol)
> 

?...
please, make sure to stick to one topic within the discussion, i'm not sure what this is about nor how it related with the ability to see NPC's and Player status (something that the engine is already capable to do for you). as for the hospital, unique quest triggers and such, you can already do this with player variables i believe right?

**Reply by LunamOdium** [reply]
I think you're misunderstanding me. Status effects; like Blind, Invulnerable, Sleep, Snare, etc. 

As far as I'm aware you cannot use a event conditional branch to see whether or not a player or NPC currently has a status effect active. Returning a value as true/false if a selected NPC or the Player has a specific effect active will allow for more variation in building events.

As an example:
Let's say I cast a spell to give an NPC the "Snare" status effect and then kill the NPC while the effect is active on them.
I want to then remove an item from the killer's(me, in this case) inventory and replace it with a different item.

basically:
@Cond. Branch: NPC has Status Effect - Snare:
   >Remove Item(itemname)
   >Add Item(itemname2)

Hope this makes sense.
And if you CAN do this already, please tell me how because I have not seen anything relating to this in the editor.


Edit: that was on topic, as one could easily create there own status effects such as Vampire Bite/Lycan Bite that after so long uncured, triggers an event to transform you into a vampire/werewolf, etc...

**Reply by WeylonSantana** [reply]
this idea is great, you can make several cool effects with it, I don't know if the event is activated before or after the statuses are cleared on death, but the concept is super interesting, if I had the time I would like to do it, but maybe will wait some long time

**Reply by LunamOdium** [reply]
Thanks! I had a go at trying to implement this myself but I'm way outta the loop with this code lol. 
It doesn't seem like it would be hard, but Idk lol
No worries, no rush! 😃

**Reply by WeylonSantana** [reply]
it's not difficult, precisely, but for now that there is no time it's difficult, but I intend to take it

**Reply by Arufonsu** [reply]
> I think you're misunderstanding me. Status effects; like Blind, Invulnerable, Sleep, Snare, etc.
> 
ohh you are right, my bad 🤣 

> Returning a value as true/false if a selected NPC or the Player has a specific effect active will allow for more variation in building events.
> 
>basically: @cond. Branch: NPC has Status Effect - X:
> 
> Do something.

that's actually pretty interesting 🆒 👍🏽
